### ISSUE TITLE
clear_output message should support clearing last output_area

### ISSUE THREAD

**Issue by lbustelo** [jupyter/notebook#1132]
Currently, `clear_output` message will ultimately clear all the `output_area` under a cell. It would be beneficial to have an option on the message so that it clear the last `output_area` div generated under the cell. For example:

If we have the following structure under a cell due to a series of `display_data` messages:
```
<div class="output">
  <div class="output_area">First </div>
  <div class="output_area">Second</div>
  <div class="output_area">Third</div>
  <div class="output_area">Fourth</div>
</div>
```

Sending a `clear_output` with a special flag (lets call it `incremental`) will yield:
```
<div class="output">
  <div class="output_area">First </div>
  <div class="output_area">Second</div>
  <div class="output_area">Third</div>
</div>
```

The limitation of `clear_output` is better appreciated when using jupyter-incubator/declarativewidgets. Since declarativewidgets are renderd in the `output_area`, taking an action that calls a function that has `display.clear_output()` will end up erasing the widgets as well. I would think that once `ipywidgets` move their content into `output-area`, they will encounter scenarios where they are affected by this. 

![clearoutput](https://cloud.githubusercontent.com/assets/2474841/13256905/2b978e6c-da12-11e5-8b2c-540903bd103e.gif)

/cc @jdfreder @SylvainCorlay

**Comment by jdfreder** [comment]
@lbustelo as a workaround you should be able to use the OutputWidget to render the outputs and control clearing.  When used as a context manager it captures all display and clear output messages.

That said, I think I'm +1 on this idea because it may help with the simple animation case cc @jasongrout

**Comment by dsblank** [comment]
I've also needed to be able to clear a different one than the last one. If a new interface had the ability to clear specific ones, that would be handy.

**Comment by takluyver** [comment]
Having addressable output widgets seems like a better approach to this than removing individual outputs by an index.

**Comment by dsblank** [comment]
Yes, and a full API would allow querying to see what addressable output items exist.

**Comment by Spandan-Madan** [comment]
Here's a workaround:-

display(button)
    def clicked(b):
         # Compute what button should display #
     
    clear_output()
    # Display computed stuff #
    display(button)`

**Comment by jasongrout** [comment]
ipywidgets also now has an OutputWidget, so you can clear just a specific set of outputs: http://ipywidgets.readthedocs.io/en/stable/examples/Output%20Widget.html
